### ISSUE TITLE
Godot 2 to 3 exporter should rename translation files (.xl -> .translation)

### ISSUE THREAD

**Issue by TheSHEEEP** [godotengine/godot#16684]
**Godot version:**
3.0

**OS/device including version:**
Windows 10

**Issue description:**
Porting my project from 2.1.4 to 3.0, I am now working on the translations.
My understanding was that there was no change in the translation system, other than changing the locale working automatically during runtime instead of having to reload scenes for new translations to apply.

But it seems now the translations do not work at all.

I have prepared a CSV file with 35 entries in three languages, and according to the output I am getting in the editor they are imported correctly:

>     compressing keys: 35
>     total compressed string size: 794 (1105 uncompressed).
>     bucket table size: 752
>     hash table size: 188
>     total collisions: 11
>     compressing keys: 35
>     total compressed string size: 971 (1185 uncompressed).
>     bucket table size: 752
>     hash table size: 188
>     total collisions: 11
>     compressing keys: 35
>     total compressed string size: 197 (198 uncompressed).
>     bucket table size: 752
>     hash table size: 188
>     total collisions: 11

I set the locale in code like this:
`TranslationServer.set_locale("en")`
And yes, that is one of the locales in the csv ;)

However, when I then try to retrieve any translation for a key via tr(), I just get the key itself returned instead of the translation. Same with using TranslationServer.translate().

What this leads to is that all the buttons and texts I have in menus show their key ("button_clickme") instead of the translation ("Click Me!").

I'm fairly certain this is a bug as I don't know what could be done differently.

**Comment by volzhs** [comment]
would you try this example project?
https://github.com/godotengine/godot-demo-projects/tree/master/gui/translation

**Comment by TheSHEEEP** [comment]
That demo works. What could be the problem, then?

**Comment by TheSHEEEP** [comment]
I put my CSV into the demo project and it works just fine there. NOW I'm confused...
Must be something weird with my project, then.

**Comment by volzhs** [comment]
reproducible sample project will help much.

**Comment by TheSHEEEP** [comment]
Can't do that, not an open source project.
But I found the problem. In my project file, this is how the translations were written:

```
[locale]
translations=PoolStringArray( "res://assets/UI/translations/translations.en.xl", "res://assets/UI/translations/translations.de.xl", "res://assets/UI/translations/translations.fi.xl" )
```

In the demo project:
```
[locale]
translations=PoolStringArray( "res://text.en.translation", "res://text.es.translation", "res://text.ja.translation" )
```

Not the different file endings. Also, I found that I do get an error when **loading** my project saying unable to load "res://assets/UI/translations/translations.fi.xl" resource... 
So what I did is completely remove that locale entry from my project.godot, add the correct translation files via project settings and it worked.

What happened is that the correct translation files were never loaded - most likely the format changed in 3.0?

In the end, this was a bug with the 2.1.4 -> 3.0 project exporter. Maybe it would be better to either fix the exporter or remove it completely. I had other problems with it before and in hindsight it would've been faster re-creating the project from scratch.

**Comment by akien-mga** [comment]
This can be implemented here, as done for `.fnt` -> `.font`: https://github.com/godotengine/godot/blob/2.1/editor/editor_export_godot3.cpp#L2222-L2225

There are a few other extensions that changed in #9201 and which also need to be converted.

**Comment by KidRigger** [comment]
I'd like to solve this. It's similar to how the font is changed right?

**Comment by sudoio** [comment]
Shouldn't this be closed since #16698 is now merged?

**Comment by akien-mga** [comment]
Indeed, fixed by #16698.
